### ISSUE TITLE
24.0.3 is ignoring statement delimiter for for certain commands in Snowflake queries.

### ISSUE THREAD

**Issue by knorman-ac** [dbeaver/dbeaver#26447]
### Description

As of yesterday, my DBeaver Community (v 24.0.3) is ignoring statement delimiters.  Trying to run any multi-statement query against Snowflake results in:

```
SQL Error [8] [0A000]: Actual statement count 3 did not match the desired statement count 1.
  Actual statement count 3 did not match the desired statement count 1.
```

In preferences, the statement delimited is set correct.  I also tried restore defaults and removing it and resetting it but to no avail.

I found that simple statements like:
```
select 1;
select 2;
```

will run as multiple statements but my code usually starts with statements such as:

```
use role abc_admin;
use database abc_db;
use schema common;
```

And this will generate the error.  It seems that "use" statements are not liked now.

I presume this is a 24.0.3 bug as that updated yesterday.
![2024-04-23_09-44-12](https://github.com/dbeaver/dbeaver/assets/135243394/c02d1d6a-f418-4252-a872-9a5fe6ee460b)
![2024-04-23_09-43-17](https://github.com/dbeaver/dbeaver/assets/135243394/fb4782f0-2873-4e8d-abe3-cec1ba8973ac)
sele

### DBeaver Version

DBeaver Community Edition 24.0.3

### Operating System

MacOS 14.4.1

### Database and driver

Database: Snowflake
Driver: snowflake-jdbc-3.14.5.jar

### Steps to reproduce

1. Connect to Snowflake
2. Open a SQL statement that begins multiple lines of "use" for example:
use role abc_admin;
use database abc_db;
use schema public;
3. Click on Execute SQL Script icon.

**Comment by knorman-ac** [comment]
I downgraded from 24.0.3 to 24.0.2 and multi-statement execution is working again.

**Comment by E1izabeth** [comment]
Thank you for bug report
You could try switching `Blank line is statement delimiter` setting to `Always`
![image](https://github.com/dbeaver/dbeaver/assets/28875055/be8457a0-d62c-4acb-ac2f-67ce518cf3f5)

**Comment by AHead-SoftwareDev** [comment]
I am having a similar/same problem but with Ubuntu 22.04 LTS and mariadb  Ver 15.1 Distrib 10.6.16-MariaDB, for debian-linux-gnu (x86_64).  Since upgrading from dbeaver-ce 24.02 to 24.03, I am unable to execute the LOAD DATA LOCAL INFILE command and I am unable to execute a stored procedure.

The LOAD DATA LOCAL INFILE generates the following error message:  

```
org.jkiss.dbeaver.model
Error
Tue Apr 23 13:44:05 PDT 2024
You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'LOAD DATA LOCAL INFILE

java.sql.SQLException: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'LOAD DATA LOCAL INFILE 
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1683)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1545)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1508)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:256)
	at org.mariadb.jdbc.MariaDbStatement.executeInternal(MariaDbStatement.java:357)
	at org.mariadb.jdbc.MariaDbStatement.execute(MariaDbStatement.java:501)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:330)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:131)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:614)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$2(SQLQueryJob.java:505)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:194)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:524)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.run(SQLQueryJob.java:239)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:115)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```
The stored procedure generates the following error message:

```
org.jkiss.dbeaver.model
Error
Tue Apr 23 13:55:19 PDT 2024
You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'DELIMITER |
BEGIN NOT ATOMIC
java.sql.SQLException: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'DELIMITER |

BEGIN NOT ATOMIC

	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1683)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1545)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1508)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:256)
	at org.mariadb.jdbc.MariaDbStatement.executeInternal(MariaDbStatement.java:357)
	at org.mariadb.jdbc.MariaDbStatement.execute(MariaDbStatement.java:501)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:330)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:131)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:614)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$2(SQLQueryJob.java:505)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:194)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:524)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.run(SQLQueryJob.java:239)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:115)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```
I have been able to run both the LOAD statement and the stored procedure for months without any problems.  The above errors did not occur until I upgraded today from DBeaver-CE 24.02 to 24.03.  I have since downgraded to 24.02 and all is well again.

As shown above, the problem appears to be with the plugin org.jkiss.dbeaver.model.

**Comment by CComparon** [comment]
Same (using MySQL). Very annoying - I now have to make a text selection before executing.

**Comment by knorman-ac** [comment]
> Thank you for bug report You could try switching `Blank line is statement delimiter` setting to `Always` ![image](https://private-user-images.githubusercontent.com/28875055/324436161-be8457a0-d62c-4acb-ac2f-67ce518cf3f5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5MTk0NDksIm5iZiI6MTcxMzkxOTE0OSwicGF0aCI6Ii8yODg3NTA1NS8zMjQ0MzYxNjEtYmU4NDU3YTAtZDYyYy00YWNiLWFjMmYtNjdjZTUxOGNmM2Y1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI0VDAwMzkwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRjYTZkOWI2MTVlOGMwMzdhZDMzOGVkMWNjMTg1NDVkYjgxZmQ1YzdhOGM2MzBjOWMyNmU3YmZkMzRkMzUyZmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vQvbouZOSMB8sn2YJHFE4oWdzP7j8krYtyUiEAqywxQ)

I reinstalled 24.0.3 and tried again.  Again it failed with the error when `Blank line is statement delimiter `was set to `Smart`.

Changing this setting to "Alway" and testing again resolved the issue.

Thank you for your suggestion.  I am able to use 24.0.3 again.

It would seem that there is a bug in the Smart logic for `Blank line is statement delimiter`.

**Comment by cluisquijada** [comment]
I had the same issue with SQL Server / Sybase connections and I needed to change the statement delimiter to 'go' and blank line is statement delimiter to 'Always' in order to 24.0.3 to work. Will this be resolved in a new release or is this the way it will be working from now onwards? Or do I need to file a bug for T-SQL based connections?

**Comment by tsjam** [comment]
Same issue here with Oracle DDL statements of this format:

```
create table MYTAB (
     ID                                NUMBER(16)     Not Null 
	,ATTR2              NUMBER(3)      Not Null 
	,CONSTRAINT MYTAB_PK PRIMARY KEY (ID) USING INDEX TABLESPACE TABSPACE1
    ,CONSTRAINT MYTAB_UQ UNIQUE (ATTR2) USING INDEX TABLESPACE TABSPACE1
    ,CONSTRAINT MTM_DOUBLE_LEVY_TRCOMPL_FK_TC FOREIGN Key (ATTR2) REFERENCES OTHERTAB (VALUE) 
)
pctfree 0 storage (initial 5m next 5m pctincrease 0) tablespace TABSPACE1 logging;

comment on table  MYTAB  is 'Table description';
comment on column MYTAB.ID                   is 'primary key';
comment on column MYTAB.ATTR2 is 'some other key';
```

In 24.0.2 it works just fine.

**Comment by E1izabeth** [comment]
The fix will be included in DBeaver 24.0.4.
Also you can try Early Access version with fix https://dbeaver.io/files/ea/ now.

**Comment by AHead-SoftwareDev** [comment]
After uninstalling DBeaver 24.0.3, installing 24.0.2, and then accidentally upgrading to DBeaver  24.0.3.202404211624 released on April 22, 2024 and going to `Windows --> Preferences --> Editors --> SQL Editor --> SQL Processing` and changing **Blank line is statement delimiter** from _Auto_ to _Always_ in the **Delimiters group** (whew), SQL file importing and stored procedures appear to be working as they did under 24.0.2.  Also, the **Statements delimiter** in the **Delimiters group** is set to _semicolon (;)_.  Don't know why the initial upgrade broke the file importing and execution of stored procedures.
